scripts/nn/test/grad_check.dml

```diff
@@ -1844,6 +1847,52 @@ softmax = function() {
   }
 }
 
+softmax2d = function() {
+  /*
+   * Gradient check for the softmax layer.
+   */
+  print("Grad checking the softmax layer with L2 loss_2d.")
+
+  # Generate data
+  N = 3 # num examples
+  C = 10 # num classes
```

Comment:
small: can you add an extra space to each of these lines so that there are two spaces before the `#` instead of one?  Python style requires two spaces at the end of the line before any inline comments.